src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/Worker.java

```diff
@@ -1163,20 +1209,31 @@ private static ExecutorService getExecutorService() {
         }
     }
 
-    private PeriodicShardSyncStrategy createPeriodicShardSyncStrategy(IKinesisProxy kinesisProxy,
-            ILeaseManager<KinesisClientLease> leaseManager) {
-        return new PeriodicShardSyncStrategy(
-                new PeriodicShardSyncManager(config.getWorkerIdentifier(), leaderDecider,
-                        new ShardSyncTask(kinesisProxy, leaseManager, config.getInitialPositionInStreamExtended(),
-                                config.shouldCleanupLeasesUponShardCompletion(),
-                                config.shouldIgnoreUnexpectedChildShards(), SHARD_SYNC_SLEEP_FOR_PERIODIC_SHARD_SYNC,
-                                shardSyncer, null), metricsFactory));
+    private PeriodicShardSyncStrategy createPeriodicShardSyncStrategy() {
+        return new PeriodicShardSyncStrategy(createPeriodicShardSyncManager());
     }
 
     private ShardEndShardSyncStrategy createShardEndShardSyncStrategy(ShardSyncTaskManager shardSyncTaskManager) {
         return new ShardEndShardSyncStrategy(shardSyncTaskManager);
     }
 
+    private PeriodicShardSyncManager createPeriodicShardSyncManager() {
+        if (leaderDecider == null) {
+            leaderDecider = new DeterministicShuffleShardSyncLeaderDecider(leaseCoordinator.getLeaseManager(),
+                    Executors.newSingleThreadScheduledExecutor(), PERIODIC_SHARD_SYNC_MAX_WORKERS_DEFAULT);
+        }
+
+        // PeriodicShardSyncManager requires a non-blank workerId. If one is not present, use a random UUID.
+        final String workerId = StringUtils.isEmpty(config.getWorkerIdentifier())
+                ? UUID.randomUUID().toString() : config.getWorkerIdentifier();
```

Comment:
When would config.getWorkerIdentifier() be empty? We used to not have this check. Is this some cases introduced by our change?

Reply:
This is set by the customer, so it's possible for it to be null or empty (isEmpty checks both). The unit tests were breaking with a null workerId so I added this - not sure why this wasn't the case before, I'll look into this further.

https://github.com/awslabs/amazon-kinesis-client/blob/v1.x/src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/Worker.java#L1166

Reply:
Discussed offline, will modify the unit tests rather than setting random workerId. PeriodicShardSyncManager will fail to initialize with null or empty workerId, but that's consistent with the current behavior.